### ISSUE TITLE
Training on a different dataset

### ISSUE THREAD

**Issue by siinem** [DrSleep/tensorflow-deeplab-resnet#144]
I want to run the train.py for a different dataset which has 38 classes. I have made the changes below:

 - Changed NUM_CLASSES in train.py to 38
 - At train.py I changed this text:

 ```
     # Saver for storing checkpoints of the model.
     saver = tf.train.Saver(var_list=tf.global_variables(), max_to_keep=10)
     # Load variables if the checkpoint is provided.
     if args.restore_from is not None:
         loader = tf.train.Saver(var_list=restore_var)
         load(loader, sess, args.restore_from)
 ```
 with the text below:
 ```
     # Saver for storing checkpoints of the model.
     saver = tf.train.Saver(var_list=restore_var, max_to_keep=40)
     # Load variables if the checkpoint is provided.
     vars_restore_gist = [v for v in tf.global_variables() if not 'fc' in v.name]  # Restore everything but last layer
     if args.restore_from is not None:
         loader = tf.train.Saver(var_list=vars_restore_gist)
         load(loader, sess, args.restore_from)
 ```
 - I have added new color labels and changed num_classes to 38 at utils.py

However, I got this error:

> NotFoundError (see above for traceback): Tensor name "res2a_branch1/weights/Momentum" not found in checkpoint files ./deeplab_resnet.ckpt
> [[Node: save_1/RestoreV2_419 = RestoreV2[dtypes=[DT_FLOAT], _device="/job:localhost/replica:0/task:0/device:CPU:0"](_arg_save_1/Const_0_0, save_1/RestoreV2_419/tensor_names, save_1/RestoreV2_419/shape_and_slices)]]
> [[Node: save_1/RestoreV2_559/_143 = _Recvclient_terminated=false, recv_device="/job:localhost/replica:0/task:0/device:GPU:0", send_device="/job:localhost/replica:0/task:0/device:CPU:0", send_device_incarnation=1, tensor_name="edge_1394_save_1/RestoreV2_559", tensor_type=DT_FLOAT, _device="/job:localhost/replica:0/task:0/device:GPU:0"]]

May you have any opinion on how i can fix it? I would be appreciated for any suggestion,

**Comment by JustWon** [comment]
according to the error message, it seems the script couldn't find the deeplab_resnet.ckpt. 
You can find the pre-trained ckpt file via following the readme instruction or [here](https://drive.google.com/drive/folders/0B_rootXHuswsZ0E4Mjh1ZU5xZVU).

**Comment by siinem** [comment]
Hi, 
when i changed the tf.global_variables()  to tf.trainable_variables(), it worked ! :)
